assembl/static2/js/app/reducers/adminReducer/index.js

```diff
@@ -63,10 +49,47 @@ export const thematicsById: ThematicsByIdReducer = (state = Map(), action) => {
       titleEntries: List(),
       video: null
     });
-    return state.set(action.id, emptyThematic.set('id', action.id));
+    const order = state.size + 1.0;
+    return state.set(action.id, emptyThematic.set('id', action.id).set('order', order));
   }
   case 'DELETE_THEMATIC':
     return state.setIn([action.id, '_toDelete'], true);
+  case 'MOVE_THEMATIC_UP': {
+    let newState = state;
+    let thematicsInOrder = state
+      .filter(thematic => !thematic.get('_toDelete'))
+      .sortBy(thematic => thematic.get('order'))
+      .map(thematic => thematic.get('id'))
+      .toList();
+    const idx = thematicsInOrder.indexOf(action.id);
+    thematicsInOrder = thematicsInOrder.delete(idx).insert(idx - 1, action.id);
+    let order = 1;
+    thematicsInOrder.forEach((thematicId) => {
+      if (newState.getIn([thematicId, 'order']) !== order) {
+        newState = newState.setIn([thematicId, 'order'], order).setIn([thematicId, '_hasChanged'], true);
+      }
+      order += 1;
+    });
+    return newState;
+  }
+  case 'MOVE_THEMATIC_DOWN': {
+    let newState = state;
+    let thematicsInOrder = state
+      .filter(thematic => !thematic.get('_toDelete'))
+      .sortBy(thematic => thematic.get('order'))
+      .map(thematic => thematic.get('id'))
+      .toList();
+    const idx = thematicsInOrder.indexOf(action.id);
+    thematicsInOrder = thematicsInOrder.delete(idx).insert(idx + 1, action.id);
+    let order = 1;
+    thematicsInOrder.forEach((thematicId) => {
+      if (newState.getIn([thematicId, 'order']) !== order) {
+        newState = newState.setIn([thematicId, 'order'], order).setIn([thematicId, '_hasChanged'], true);
+      }
+      order += 1;
```

Comment:
shouldn't this one be -= 1 ?

Reply:
no, we iterate here over the thematics and reset the order incrementaly.